source/MaterialXCore/Element.h

```diff
@@ -42,6 +42,27 @@ using ElementMap = std::unordered_map<string, ElementPtr>;
 /// A standard function taking an ElementPtr and returning a boolean.
 using ElementPredicate = std::function<bool(ElementPtr)>;
 
+/// @class CopyOptions
+/// A set options for controlling the behavior of copying of elements.
+class CopyOptions
+{
+  public:
+    CopyOptions() :
+        skipDuplicateElements(false),
+        sourceUris(false)
+    {
+    }
+    ~CopyOptions() { }
+
+    /// If true, elements at the same scope with duplicate names will be skipped;
+    /// otherwise, they will trigger an exception.  Defaults to false.
+    bool skipDuplicateElements;
+
+    /// If true, then source URIs from the given element
+    /// and its descendants are also copied.  Defaults to false.
+    bool sourceUris;
```

Comment:
I would recommend naming this data member ```copySourceUris```, for additional clarity and consistency with ```skipDuplicateElements```.